scraper.rb

```diff
@@ -33,6 +36,67 @@ def tidy
   end
 end
 
+class Mirror
+  include Capybara::DSL
+
+  attr_accessor :url
+
+  def save_page(url)
+    url = url.match(/(.*)_piref[\d_]+\.(next_page.*)/).captures.join('')
+    sha = Digest::SHA1.hexdigest url
+    path = File.join('.', CACHE_DIR, sha)
+
+    return if File.exist?(path)
+
+    File.open(path,"w") do |f|
+      f.write(page.html)
+    end
+    sleep(1)
```

Comment:
Er, because the site has seemed quite fragile in the past so I was being cautious about sending lots of requests. And because it's polite :) This may be old fashioned of me.